.azure-pipelines/dependency_check/dependency_check.py

```diff
@@ -0,0 +1,189 @@
+            for node in ast.walk(tree):
+                # # Check for `import` statements
+                if isinstance(node, ast.Import):
+                    for entry in node.names:
+                        imported_modules_in_files[file_path].append({
+                            "type": "import",
+                            "module": entry.name,
+                            "module_path": get_module_path(entry.name),
+                            "asname": entry.asname
+                        })
+                # Check for `from ... import ...` statements
+                if isinstance(node, ast.ImportFrom):
+                    for entry in node.names:
+                        imported_modules_in_files[file_path].append({
+                            "type": "from_import",
+                            "module": node.module,
+                            "module_path": get_module_path(node.module, node.level, file_path),
+                            "alias": entry.name,
+                            "asname": entry.asname
+                        })
+    return imported_modules_in_files
+
+
+def get_feature_path(path):
+    """
+        For our repo, we can consider the folders like "acl", "bgp" as feature folders.
+        In this function, we will retrieve the path of the top-level feature directories.
+        In other words, we will retrieve the absolute paths of the first-level folders under `sonic-mgmt/tests`
+    """
+    if path is None:
+        return None
+
+    file_path = os.path.abspath(path)
+    target_path = "tests"
+    index = file_path.find(target_path)
+
+    if index != -1:
+        project_path = file_path[:index + len(target_path)]
+    else:
+        return None
+
+    feature = file_path[len(project_path) + 1:].split("/")[0]
+    return os.path.join(project_path, feature)
+
+
+def check_cross_dependency(imports_in_script):
+    """
+        Check if there are cross-feature dependency in each file.
+    """
+    cross_dependency = False
+    for file_path, imported_modules in imports_in_script.items():
+        file_feature_path = get_feature_path(file_path)
+        for imported_module in imported_modules:
+            imported_module_feature_path = get_feature_path(imported_module["module_path"])
+            if imported_module_feature_path is not None:
+                project_path = os.path.dirname(file_feature_path)
+                # Import from these paths are allowed.
+                if imported_module_feature_path not in [os.path.join(project_path, "common"),
+                                                        os.path.join(project_path, "ptf_runner.py"),
+                                                        os.path.join(project_path, "conftest.py"),
+                                                        file_feature_path]:
+                    print("There is a cross-feature dependence. File: {}, import module: {}"
+                          .format(file_path, imported_module["module"]))
```

Comment:
Here may have duplicate prints, how about just print 1 line and continue, or you can also print the alias

Reply:
I think one script one line is more clearly for user to read.